lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,436 @@
+            Text(errorText!, style: textTheme.caption?.copyWith(color: theme.destructiveSwatch))
+                .withOnlyPadding(context, left: Spacing.small)
+          ] else if (helperText != null) ...[
+            context.verticalBox(Spacing.xxxSmall),
+            Text(helperText!, style: textTheme.caption?.copyWith(color: theme.neutralSwatch.shade400))
+                .withOnlyPadding(context, left: Spacing.small)
+          ]
+        ],
+      ),
+    );
+  }
+}
+
+/// Rich text editor modal.
+///
+/// Holds the rich text editor using `flutter_quill` library.
+/// Places [_RichTextFieldToolbar] - a toolbar of actions to customize the text content - above the keyboard when it's
+/// visible.
+class _RichTextFieldModal extends HookConsumerWidget {
+  const _RichTextFieldModal({required this.title, required this.controller, required this.placeholder, this.focus});
+
+  final Widget title;
+  final quill.QuillController controller;
+  final String placeholder;
+  final FocusNode? focus;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+
+    final focusNode = focus ?? useFocusNode();
+
+    final isFocused = useState(focusNode.hasFocus);
+    final hasSelection = useState(controller.selection.start != controller.selection.end);
+
+    useEffect(() {
+      void focusUpdate() => isFocused.value = focusNode.hasFocus;
+      void selectionUpdate() => hasSelection.value = controller.selection.start != controller.selection.end;
+
+      focusNode.addListener(focusUpdate);
+      controller.addListener(selectionUpdate);
+
+      return () {
+        focusNode.removeListener(focusUpdate);
+        controller.removeListener(selectionUpdate);
+      };
+    });
+
+    final editor = _ThemedEditor(
+      controller: controller,
+      placeholder: placeholder,
+      backgroundColor: theme.neutralSwatch.shade800,
+      focus: focusNode,
+    );
+
+    final okButton = CustomTextButton(
+      text: strings.ok.toUpperCase(),
+      onPressed: Navigator.of(context).pop,
+    );
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              Align(alignment: Alignment.centerRight, child: okButton),
+              context.verticalBox(Spacing.xxxSmall),
+              title,
+              context.verticalBox(Spacing.xxSmall),
+              Expanded(child: editor),
+              context.verticalBox(Spacing.xxSmall),
+            ],
+          ).withSymmetricalPadding(context, horizontal: Spacing.medium),
+        ),
+        // Only show toolbar action items when the field has focus or when the user has an active selection.
+        if (isFocused.value || hasSelection.value) ...[
+          _RichTextFieldToolbar(controller),
+          SizedBox(height: MediaQuery.of(context).viewInsets.bottom),
```

Comment:
Possibly explain that this is a way to add a bottom space that takes the same height of the currently visible keyboard